bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -494,6 +494,10 @@ public synchronized boolean isClosed() {
         return getLedgerMetadata().isClosed();
     }
 
+    boolean isHandleWritable() {
+        return !getLedgerMetadata().isClosed();
```

Comment:
How do we deal with readonly ledger handle?

Reply:
The ledgerMetadata is `closed` for a readonly ledger handle.

Reply:
I am not sure that it is 'closed' for a non recovery reader.

Maybe I need a coffee :)

Reply:
pr #1804 and update the PR here using same approach.

Reply:
> The ledgerMetadata is closed for a readonly ledger handle.

@sijie  is that always true? Can't we get readonly handle for an active ledger? We do.

Reply:
@jvrao that was my concern as well, but it turned out that changing this method to return false for ReadOnlyLeadgerHandle would break many recovery tests